### PR TITLE
SpinButton updates from accessibility testing

### PR DESCRIPTION
**Pull request by spmonahan**
## Current Behavior

1. SpinButton has some styling issues in Windows High Contrast Mode
2. Increment and decrement buttons do not have labels

## New Behavior

1. Styling has been added for address the WHCM issues. A follow up PR will be made to make icons do this be default.
2. A `strings` prop has been added to SpinButton allowing labels be applied to the increment and decrement buttons. These strings support a token, `{step}`, that will be replaced with the value of the `step` prop when present in the string. This allows the labels to be localized as needed. By default the English localization is included.

## Related Issue(s)

#22307

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                      |     Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                                                                    Change |
| :----------------------------------------------------------------------------------------------------- | ---------------------------: | --------------------------: | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr> | `41.512 kB`<br />`11.912 kB` | `42.72 kB`<br />`12.187 kB` | `1.208 kB` <img aria-hidden="true" src="https://microsoft.github.io/sizeAuditor-website/images/icons/IncreaseYellow.svg" /><br />`275 B` <img aria-hidden="true" src="https://microsoft.github.io/sizeAuditor-website/images/icons/IncreaseYellow.svg" /> |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/543b0c6b1e1d950b454c43662fa3e7b911139919'>543b0c6b1e1d950b454c43662fa3e7b911139919</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx [hunk 1]
```diff
@@ -34,6 +36,16 @@ const MAX_SPIN_TIME_MS = 1000;
...
 
+// TODO: discuss this with design
+// const selectEntireRange = (input: HTMLInputElement | null): void => {
+//   if (!input) {
+//     return;
+//   }
+
+//   input.focus();
+//   input.setSelectionRange(0, input.value.length);
+// };
```
Comment (khmakoto) [inline packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx:47]: Should this stay here? Seems a bit unrelated to the other changes in this PR.

File: packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx [hunk 2]
```diff
@@ -151,6 +169,13 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
     setTextValue(newTextValue);
   }, [value, displayValue, currentValue, precision, setAtBound, min, max]);
 
+  // TODO: see TODO at the top of this file for selectEntireRange
+  // React.useEffect(() => {
+  //   if (spinState !== 'rest') {
+  //     selectEntireRange(inputRef.current);
+  //   }
+  // }, [textValue, spinState]);
```
Comment (khmakoto) [inline packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx:177]: Same comment as above.

File: packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx (hunk 1 above)
Comment (spmonahan) [reply packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx:47]: I added after discovering a bug in NVDA where it announces "blank" when spinning the value via the keyboard. @smhigley suggested I try selecting all the text in the input to get  NVDA to announce it but that didn't pan out. It's left here because it might still be a nice UX but I wanted to discuss it with design first. Happy to remove it to another branch if you want. LMK.

Comment (spmonahan) [reply packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx]: Decided to just remove these and keep em around in another branch.

**Review by sopranopillow — APPROVED** [review]
LGTM

**Review by khmakoto — APPROVED** [review]
